ts/client/scripts/risk.ts

```diff
@@ -185,18 +196,27 @@ async function computePriceImpactForLiqor(
     // liabs by the liqor, who would immediately want to reduce to 0
     // Assuming assets need to be sold to USDC
     const assets = mangoAccountsWithHealth.reduce((sum, a) => {
-      if (bank.maintAssetWeight.eq(ZERO_I80F48())) {
-        return sum.add(ZERO_I80F48());
-      }
       // How much would health increase for every unit liab moved to liqor
+      // assetprice * (liabweight/(1+liabliqfee) - assetweight)
       const tokenAssetHealthContrib = ONE_I80F48()
         .sub(bank.liquidationFee)
         .mul(bank.price)
-        .mul(bank.maintAssetWeight);
+        .mul(bank.initAssetWeight);
+      bank.price.mul(
+        Array.from(group.banksMapByTokenIndex.values())
+          .flat()
+          .map((bank) => bank.initLiabWeight)
+          .reduce((prev, curr) => (prev.lt(curr) ? prev : curr))
+          .div(ONE_I80F48().add(bank.liquidationFee))
+          .sub(bank.initAssetWeight),
+      );
       // Abs collateral/asset
       const maxTokenHealthAsset = a.account
         .getTokenBalance(bank)
         .max(ZERO_I80F48());
+      if (bank.initAssetWeight.eq(ZERO_I80F48())) {
+        return sum.add(maxTokenHealthAsset);
```

Comment:
why would all asset be added? shouldn't that still depend on the health and how much health is gained by liquidating the assetweight=0 token to offset a borrow?

Reply:
correct, with the change in equation actually this condition has become redundant, removed